### DISCUSSION TITLE
DS3502, Digital Poti, Device found, but no reaction.  Try on Arduino is o.k.

### DISCUSSION THREAD

**Discussion by rudikx6** [arendst/Tasmota#21482] [Support request] [answered]
Hey guys,

I think I need some help now. Read already many in the Web, but still in trouble.

I try to use a DS3502 digital Potentiometer with tasmota.
I was able to compile the Firmware with the DS3502 part and flashed it to a ESP2866 on a NodeMCU v3.
(v14.0.0. with Tasmota_DE as basic)
In the WebUI I configure the I2C Pins and activate the I2Cdriver67.

While booting I can see the ESP was able to find the DS3502. It shows a message in the consol. Something linke… “DS3502 found, at address 0x28”
And I check the active drivers with “I2CDriver” and got response …(7,67)

Now I think everything is fine and try to set the “Wiper” to a value.
For example iam using command in WebConsole “Wiper 90” or “Wiper1 90” 
And from console it looks good because I get RESULT …{Wiper1, 90}
If the sensor is not connected, i get negativ result. so i think the communication is o.k.

So far so good 😊
But in Hardware on the DS3502, nothing happened!...

My first intention was that my schematic on DS3502 side was buggy…. So I tried the Basic Examlpe from Adrafruit with an Arduino. And it works. So on DS3502 no Problem.
What iam doing wrong?

I read in the Commands Docs something about this:
POTI 0..127 = set POTI for wiper x
STATUS = get wiper position for wiper x
RESET = reset settings for wiper x

Do i have to user the POTI command?

And on Github I found a comment from arendst, “Four DS3502 devices have been implemented today. Use command Wiper<x> 0..127 to control them.”
Its from May 2022. 

And in the list of supported I2C Devices Is something written about Driver “xdrv_61” this gets me confused, because I think driver 67 was correct.
And I was not able to activate the 61 by “I2CDriver61 1”
Is the I2CDriver67 like a package including the xdrv_61?

Do I have some misunderstanding with the commands?

Greetings, Rudi from Germany

**Comment by HofeBY2** [comment]
Try
I2Cdriver
to see which Driver Nr. are in your build.
Then search about all those Nr about a device that also use I2C device Adress 0x28.
Then remove unwateted Driver ID and rebuild new Firmware. Or simple use command
I2CdriverNr 0

Greetings from Oberfranken
HofeBY

**Reply by barbudor** [reply]
There is 2 driver numbers - see https://tasmota.github.io/docs/I2CDEVICES/#supported-i2c-devices
- The I2C driver number on the 1st column is the one used with the `I2cDriver`  command => 67
- The general Tasmota driver number => Just the one that appear in the source code file name (61). You don't need to take care of it

Looking at the code,  don't understand the doc. the `Wiper<x>`  command seems to allow only a value between 0 and 127
There no reference to POTI, STATUS or RESET

**Reply by rudikx6** [reply]
"I2Cdriver" shows me 2 active drivers in my bulid. (7,67)
So i checked the driver7 and there is no device withe same address. but i also try to deactivate it. but still same Problem.
![2024-05-25 08_20_48-Window](https://github.com/arendst/Tasmota/assets/21043592/c78a33a4-7460-4cfe-a6b9-92ea2713ee8a)




Now i try to switch the address of the Sensor. from 0x28 to 0x29 and 0x2B, but with this addres Tasmota could not find the Device.


I also tried the i2cscan command, but it shows me everytime errors, also if i use default addres and tasmota found the device on 0x28!:
![2024-05-25 08_42_59-Window](https://github.com/arendst/Tasmota/assets/21043592/02be506d-bb15-495b-8c17-9a3b8ed13be7)


could someone take a look on my config? is it coorect?
![2024-05-25 08_47_16-● user_config_override h - Untitled (Workspace) - Visual Studio Code](https://github.com/arendst/Tasmota/assets/21043592/e14c21ca-318f-4999-bf32-c4e3f59dfb1c)

**Comment by sfromis** [comment]
I'm not using the DS3502, but checking the [driver](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_61_ds3502.ino) source code, I can see that it only has one command, `Wiper<x> value` where the `x` is when you can have more than one DS3502 on successive addresses, with the default being 1 for the first. Thus, `Wiper 90` and `Wiper1 90` is functionally the same command. The driver does not appear to have anything other than selecting which chip, and which 0..127 value to send to the chip. Not entirely sure how the docs text for status/reset is meant....

And no, I know nothing about actually using the chip from Tasmota.

When it comes to the numbers, Tasmota has separate numbering schemes for I2C devices, drivers (of which not all are I2C), sensors (including I2C), energy, lights and display (including I2C). In this case, the number 67 pertains to the `I2CDriver` command and output, while number 61 is the more general "driver" concept, and command `Driver`. I2CDriver61 pertains to `xsns_89_t67xx.ino`. Thus, the same device/driver can have 2 or 3 numbers at the same time, not to be mixed up.

Activating via the I2CDriver command is only relevant if you *already* have the driver included *and* inactivated, showing up with an `!` on the I2CDriver command output.

**Comment by rudikx6** [comment, marked as answer]
Found the Problem.... 

the error on i2cScan brings me the hint. **Missind pullUp resistors in SCL & SDA.**
I added it and now the DS3502 reacts as aspected :)

Also the I2CScan brings now no error.

my fault was to trust the test withe the Arduino. but the Arduino has build in pullups on I2C.


Thanks for the help, in the end it helped me to reach my goal.

**Reply by HofeBY** [reply]
Small, cheeky, treacherous resistances
